opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/LDAPAuthenticationManager.java

```diff
@@ -84,9 +97,17 @@ public String authenticate(String username, String password) throws CatalogAuthe
             env.put(Context.PROVIDER_URL, host);
 
             env.put(Context.SECURITY_AUTHENTICATION, "simple");
-            env.put(Context.SECURITY_PRINCIPAL, rdn);
             env.put(Context.SECURITY_CREDENTIALS, password);
 
+            if (ldaps) {
+                String uid = userInfoFromLDAP.get(0).get(dnKey).get(0).toString();
+                String dn = String.format(dnFormat, uid);
+                env.put(Context.SECURITY_PRINCIPAL, dn);
+                env.put(Context.SECURITY_PROTOCOL, "ssl");
+            } else {
+                env.put(Context.SECURITY_PRINCIPAL, rdn);
```

Comment:
The original/default behaviour here was to pass the contents of `gecos` to the auth request rather than a distinguished name like `uid=a_user,ou=people...`. Does this work?

Reply:
Looks like in other cases, the gecos is formatted correctly. In other LDAP systems though, the distinguished name may be stored in a different attribute, or just as the ID of the user object from the search:

```python
>>> import ldap
>>> l = ldap.initialize(ldap_url)
>>> l.search_s(base_dn, ldap.SCOPE_SUBTREE, '(uid=me)')
[
    (
        "uid=me,ou=people,dc=ge,dc=co,dc=uk",  # DN could be here...
        {
            "uid": "me",
            "gecos": "uid=me,ou=people,dc=ge,dc=co,dc=uk",  # ... or here (default OpenCGA behaviour)...
            "some_other_field": "uid=me,ou=people,dc=ge,dc=co,dc=uk"  # ... or here
        }
    )
]
```

So we still need some way of telling OpenCGA where to find the DN, but it may be possible to remove some of the added config options